File: backend/src/common/models/device.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Deserialize, Serialize, Clone, sqlx::FromRow, Debug)]
pub struct Device {
    pub id: i64,
    pub uuid: Uuid,
    pub description: String,
    pub area: String,
}

File: frontend/types/device.ts
export class Device {
  id: number;
  uuid: string;
  description: string;
  area: string;

  constructor(id: number, uuid: string, description: string, area: string) {
    this.id = id;
    this.uuid = uuid;
    this.description = description;
    this.area = area;
  }
}
